### ISSUE TITLE
model_to_graphviz does not draw edges to interaction nodes

### ISSUE THREAD

**Issue by jvparidon** [pymc-devs/pymc#3413]
## Problem description
`model_to_graphviz` is a great visualization tool, but it currently does not draw edges to interaction terms. Nodes with names that contain a colon are not connected to the model graph because graphviz does not understand how to draw the edges.

Calls to graphviz from PyMC3 are translated into DOT and that language standard treats colons as a special character. When the graph is drawn and the instructions for drawing the edges are parsed, having a colon in one of the node names causes graphviz to look for a non-existent node to connect the edge to. This leads to missing edges in the graph and graphviz will issue a warning, but neither graphviz nor PyMC3 throws an exception.

## Minimal working example
```python
import numpy as np
import pandas as pd
import pymc3 as pm

# toy data
df = pd.DataFrame({
    'y': np.random.random(10),
    'x1': np.random.random(10),
    'x2': np.random.random(10)
})

with pm.Model() as linear_model:
    pm.glm.GLM.from_formula('y ~ x1 * x2', data=df)
    pm.model_to_graphviz(linear_model).view()
```

## Versions and main components
* PyMC3 Version: 3.6
* Theano Version: 1.0.4
* Python Version: 3.7.1
* Operating system: macOS 10.14.1
* How did you install PyMC3: pip

**Comment by lucianopaz** [comment]
Thanks for reporting thid @JvParidon. It seems like this should be easy to fix by inserting the names in [quotes](https://stackoverflow.com/questions/18444406/how-do-i-use-special-characters-in-a-dot-file-node-id), or somehow escaping the colon in the names. Would you be interested in trying to address this with a PR?

**Comment by jvparidon** [comment]
Unfortunately, graphviz appears to parse colons in node names even if the names are in quotes, [this is a known issue.](https://github.com/xflr6/graphviz/issues/53) The recommended workaround is to just remove colons from the node names but leave them in the node labels which will be drawn in the graph.
A string replace in the node name to replace colons with some other character before passing them to graphviz does indeed work, and seems like the easiest fix to me. I'll submit a PR.

**Comment by junpenglao** [comment]
Closed in #3414
